src/main/java/brave/opentracing/BraveTracer.java

```diff
@@ -50,27 +52,65 @@
  * @see Propagation
  */
 public final class BraveTracer implements Tracer {
-
-  static final List<String> PROPAGATION_KEYS = Propagation.B3_STRING.keys();
-  static final TraceContext.Injector<TextMap> INJECTOR =
-      Propagation.B3_STRING.injector(TextMap::put);
-  static final TraceContext.Extractor<TextMapView> EXTRACTOR =
-      Propagation.B3_STRING.extractor(TextMapView::get);
-  static final Set<String> FIELDS_LOWER_CASE = lowercaseSet(PROPAGATION_KEYS);
-
-  private final brave.Tracer brave4;
-
   /**
    * Returns an implementation of {@linkplain io.opentracing.Tracer} which delegates
    * the the provided Brave Tracer.
    */
   public static BraveTracer wrap(brave.Tracer brave4) {
-    if (brave4 == null) throw new NullPointerException("brave tracer == null");
-    return new BraveTracer(brave4);
+    return newBuilder(brave4).build();
+  }
+
+  public static Builder newBuilder(brave.Tracer brave4) {
+    return new Builder(brave4);
+  }
+
+  public static final class Builder {
+    brave.Tracer brave4;
+    Map<Format<TextMap>, Propagation<String>> formatToPropagation = new LinkedHashMap<>();
```

Comment:
Since we are making this pluggable, I'm not sure it makes sense to confine this to `Format<TextMap>`  when `Format.BuiltIn.BINARY` exists and could have a custom propagation created for it